### ISSUE TITLE
[Masthead] Search icon does not have focus

### ISSUE THREAD

**Issue by jeffchew** [carbon-design-system/ibm-dotcom-library#595]
## Detailed description

> Describe in detail the issue you're having.

When keyboard navigating, the search icon in the masthead does not have a focus state

> Is this a feature request (new component, new icon), a bug, or a general
> issue?

Bug

> Is this issue related to a specific component?

Masthead

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Have focus state when using keyboard navigation to the search icon
